lib/Dialect/Torch/Transforms/DecomposeComplexOps.cpp

```diff
@@ -1791,6 +1793,56 @@ class DecomposeAtenToDtypeLayoutOp
 };
 } // namespace
 
+namespace {
+// Decompose the `aten.select_scatter` operation into `aten.slice_scatter` op.
+class DecomposeAtenSelectScatterOp
+    : public OpRewritePattern<AtenSelectScatterOp> {
+public:
+  using OpRewritePattern::OpRewritePattern;
+  LogicalResult matchAndRewrite(AtenSelectScatterOp op,
+                                PatternRewriter &rewriter) const override {
+    Location loc = op.getLoc();
+    Value start = op.index();
+    Value dim = op.dim();
+    Value self = op.self();
+    Value src = op.src();
+
+    Value one =
+        rewriter.create<ConstantIntOp>(loc, rewriter.getI64IntegerAttr(1));
+    Value startPlusOne =
+        rewriter.create<AtenAddIntOp>(loc, one.getType(), start, one);
+    BaseTensorType srcTensorType = src.getType().cast<BaseTensorType>();
+    SmallVector<int64_t> sizes;
+    int64_t dimInt = 0;
+    if (srcTensorType.hasSizes()) {
+      ArrayRef<int64_t> srcShape = srcTensorType.getSizes();
+      // `src` has a reduced rank. Hence add 1.
+      int64_t srcRank = srcShape.size() + 1;
```

Comment:
This is really just for my own understanding, but why do we add 1 here? What's the expected rank for `src` and why is it originally reduced to one less than the expected rank?

Reply:
src `dim` dimension is squeezed up. To unsqueeze it 1 is added.